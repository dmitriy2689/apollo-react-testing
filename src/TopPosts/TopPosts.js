import React from 'react';
import { Link } from 'react-router-dom'
import { Query } from 'react-apollo';
import query from './TopPosts.query';

const TopPosts = () => (
  <Query query={query}>
    {({ loading, error, data }) => {
      if (loading) return <p>Loading...</p>;
      if (error) return <p>Error :(</p>;

      return (
        <ul key='topPosts'>
          {data.allPosts.map(({ id, title, user }) => (
            <li key={id}>
              <Link to={`/post/${id}`}>
                {user ? user.name : 'incognito'}, {title ? title : '***'}
              </Link>
            </li>
          ))}
        </ul>
      );
    }}
  </Query>
);

export default TopPosts;