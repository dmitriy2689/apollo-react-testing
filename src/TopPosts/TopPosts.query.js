import gql from 'graphql-tag';

const query = gql`
  query {
    allPosts(orderBy: updatedAt_DESC, first: 7){
      id
      title
      user {
        id
        name
      }
    }
  }
`;

export default query;