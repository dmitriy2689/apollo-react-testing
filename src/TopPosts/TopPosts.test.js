import React from 'react';
import { MockedProvider } from "react-apollo/test-utils";
import renderer from "react-test-renderer";
import { shallow, mount, render } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import wait from "waait";
import TopPosts from './TopPosts';
import query from './TopPosts.query';

it('should render posts list', async () => {
  const mock = {
    request: { query },
    result: { data: { allPosts: [
      { id: "1", title: "title1", user: {id: "1", name: 'title1'} },
      { id: "2", title: "title2", user: {id: "2", name: 'title2'} }
    ]}}
  };

  const wrapper = shallow(
    <BrowserRouter>
      <MockedProvider mocks={[mock]} addTypename={false}>
        <TopPosts />
      </MockedProvider>
    </BrowserRouter>
  );

  await wait(0);

  expect(wrapper).toMatchSnapshot();
})
