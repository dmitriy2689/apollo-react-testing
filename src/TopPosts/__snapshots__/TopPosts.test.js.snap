// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render posts list 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <MockedProvider
    addTypename={false}
    mocks={
      Array [
        Object {
          "request": Object {
            "query": Object {
              "definitions": Array [
                Object {
                  "directives": Array [],
                  "kind": "OperationDefinition",
                  "name": undefined,
                  "operation": "query",
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [
                          Object {
                            "kind": "Argument",
                            "name": Object {
                              "kind": "Name",
                              "value": "orderBy",
                            },
                            "value": Object {
                              "kind": "EnumValue",
                              "value": "updatedAt_DESC",
                            },
                          },
                          Object {
                            "kind": "Argument",
                            "name": Object {
                              "kind": "Name",
                              "value": "first",
                            },
                            "value": Object {
                              "kind": "IntValue",
                              "value": "7",
                            },
                          },
                        ],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "allPosts",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "id",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "title",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "user",
                              },
                              "selectionSet": Object {
                                "kind": "SelectionSet",
                                "selections": Array [
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "id",
                                    },
                                    "selectionSet": undefined,
                                  },
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "name",
                                    },
                                    "selectionSet": undefined,
                                  },
                                ],
                              },
                            },
                          ],
                        },
                      },
                    ],
                  },
                  "variableDefinitions": Array [],
                },
              ],
              "kind": "Document",
              "loc": Object {
                "end": 136,
                "start": 0,
              },
            },
          },
          "result": Object {
            "data": Object {
              "allPosts": Array [
                Object {
                  "id": "1",
                  "title": "title1",
                  "user": Object {
                    "id": "1",
                    "name": "title1",
                  },
                },
                Object {
                  "id": "2",
                  "title": "title2",
                  "user": Object {
                    "id": "2",
                    "name": "title2",
                  },
                },
              ],
            },
          },
        },
      ]
    }
  >
    <TopPosts />
  </MockedProvider>
</Router>
`;
